src/Persistance/Json/JsonShoppingListRepository.php

```diff
@@ -90,4 +91,18 @@ private function storeAs(ShoppingList|string $slug): string
 
         return $slug . '.json';
     }
+
+    /**
+     * @param string $slug
+     * @return void
+     */
+    public function delete(string $slug): void
+    {
+        $filename = $this->storeAs($slug);
+
+        if ($this->files->exists($filename)) {
```

Comment:
Suggestion: you're checking if the file exists here, but the file handler also checks whether the file exists. So that means this check is being done twice - it would be more efficient to check whether it exists just once. There's no right answer as to whether the repo or the file handler is the best place to check if the file exists - different developers would go for different approaches. Personally I'd check in the file handler i.e. the repo can just call `$this->files->unlink()` without the `if` statement.

Reply:
Ah, that does make sense. I chose the approach you have stated above and just simply did $this->files-unlink($filename) in the repo. It makes sense that the fileHandler would check if the file exists since it handles everything that deals with the files.